server/models/huggingface_models.py

```diff
@@ -0,0 +1,97 @@
+from transformers import RobertaTokenizer, T5ForConditionalGeneration
+from transformers import AutoModelForCausalLM
+import transformers
+import torch
+
+# Does not work locally because of the size of the models.
+
+'''
+def llama_model(text: str) -> str:
+    tokenizer = AutoTokenizer.from_pretrained("daryl149/llama-2-7b-chat-hf")
+    model = AutoModel.from_pretrained("daryl149/llama-2-7b-chat-hf")
+
+
+    # tokenizer = LlamaTokenizer.from_pretrained("/output/path")
+    # model = LlamaForCausalLM.from_pretrained("/output/path")
+    tokenized = tokenizer.encode([text])
+    output = model.generate(**tokenized)
+    return output
+
+
+def codet5_model(text: str) -> str:
+    checkpoint = "Salesforce/instructcodet5p-16b"
+    device = torch.device("cuda:0") if torch.cuda.is_available() else torch.device("cpu")
+    tokenizer = AutoTokenizer.from_pretrained(checkpoint)
+    model = AutoModelForSeq2SeqLM.from_pretrained(checkpoint,
+                                                  torch_dtype=torch.float16,
+                                                  low_cpu_mem_usage=True,
+                                                  trust_remote_code=True).to(device)
+
+    encoding = tokenizer("def print_hello_world():", return_tensors="pt").to(device)
+    encoding['decoder_input_ids'] = encoding['input_ids'].clone()
+    outputs = model.generate(**encoding, max_length=15)
+    print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+'''
+
+
+def codet5_base_model(text: str):
+    tokenizer = RobertaTokenizer.from_pretrained('Salesforce/codet5-base')
+    model = T5ForConditionalGeneration.from_pretrained('Salesforce/codet5-base')
+    input_ids = tokenizer(text, return_tensors="pt").input_ids
+    generated_ids = model.generate(input_ids, max_length=8)
+    print(tokenizer.decode(generated_ids[0], skip_special_tokens=True))
+
+
+def codet5_small_model(text: str):
+    tokenizer = RobertaTokenizer.from_pretrained('Salesforce/codet5-small')
+    model = T5ForConditionalGeneration.from_pretrained('Salesforce/codet5-small')
+    input_ids = tokenizer(text, return_tensors="pt").input_ids
+    generated_ids = model.generate(input_ids, max_length=8)
+    print(tokenizer.decode(generated_ids[0], skip_special_tokens=True))
+
+
+def starcoder_model(text: str):
+    checkpoint = "bigcode/starcoderbase-1b"
+    device = torch.device("cuda:0") if torch.cuda.is_available() else torch.device("cpu")
+    tokenizer = AutoTokenizer.from_pretrained(checkpoint)
+    model = AutoModelForCausalLM.from_pretrained(checkpoint).to(device)
+    inputs = tokenizer.encode(text, return_tensors="pt").to(device)
+    outputs = model.generate(inputs)
+    print(tokenizer.decode(outputs[0]))
+
+
+def llama_model(text: str):
+    model = "codellama/CodeLlama-7b-hf"
+    tokenizer = AutoTokenizer.from_pretrained(model)
+    pipeline = transformers.pipeline(
+        "text-generation",
+        model=model,
+        torch_dtype=torch.float16,
```

Comment:
let's use `torch.bfloat16` but still that would be ~12Gb

I've used int4 quantisation

```
    nf4_config = BitsAndBytesConfig(
        load_in_4bit=True,
        bnb_4bit_quant_type="nf4",
        bnb_4bit_use_double_quant=True,
        bnb_4bit_compute_dtype=torch.bfloat16
    )
    model_nf4 = AutoModelForCausalLM.from_pretrained(args.model, quantization_config=nf4_config)
    tokenizer = AutoTokenizer.from_pretrained(args.model)

    pipeline = transformers.pipeline(
        "text-generation",
        model=model_nf4,
        torch_dtype=torch.bfloat16,
        tokenizer=tokenizer
    )
```